sdk/apis/kubermatic/v1/policy_binding.go

```
@@ -84,20 +82,40 @@ type PolicyBindingSpec struct {
 	// +kubebuilder:validation:Required
 	PolicyTemplateRef corev1.ObjectReference `json:"policyTemplateRef"`
 
-	// Enabled controls whether the policy defined by the template should be actively applied to the cluster.
+	// KyvernoPolicyNamespace specifies the Kyverno namespace to deploy the Kyverno Policy into
 	//
 	// Relevant only if the referenced PolicyTemplate has spec.enforced=false.
+	// If Template.NamespacedPolicy is true and this field is omitted, no Kyverno Policy resources will be created.
 	//
 	// +optional
-	Enabled *bool `json:"enabled,omitempty"`
+	KyvernoPolicyNamespace *KyvernoPolicyNamespace `json:"kyvernoPolicyNamespace,omitempty"`
+}
+
+// KyvernoPolicyNamespace specifies the Kyverno namespace to deploy the Kyverno Policy into
+// This is relevant only if a Kyverno Policy resource is created because Kyverno Policy is namespaced.
+// For Kyverno ClusterPolicy, this field is ignored.
+type KyvernoPolicyNamespace struct {
+	// Name is the namespace to deploy the Kyverno Policy into.
+	//
+	// +kubebuilder:validation:Pattern:=`^(|[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*)`
+	// +kubebuilder:validation:MaxLength:=63
+	// +kubebuilder:validation:Type=string
+	Name string `json:"name"`
 
-	// NamespaceSelector specifies which namespaces the Kyverno Policy resource(s) should be created in
+	// Create defines whether the namespace should be created if it does not exist. Defaults to true
 	//
-	// Relevant only if Template.NamespacedPolicy is true.
-	// If Template.NamespacedPolicy is true and this selector is omitted, no Kyverno Policy resources will be created.
+	// +kubebuilder:default:=true
+	Create bool `json:"create"`
```

I think I get where you're coming from with this, but

1. I cannot see where in the controller this field is even used.
2. Can you describe usecases you have in mind where someone would not want the namespace to be created? And how would this be handled, is the namespace not existing then causing us to requeue the binding (i.e. wait for some other process to create the namespace)? Or would we need to watch namespaces in the usercluster to react? Or is it meant as a stop mechanism ("if the namespace doesn't exists, it won't exist because I don't want this policy and this won't change")? TL;DR I am not sure this much configurability is needed here and this might just create more error states we would have to somehow handle. But maybe there is a good reason for this field?

---

I actually agree that this adds extra complexity, this struct will be used by the policy installation controller (not current controller) when it creates Kyverno Policy resources in UC.

We also have the case when a PolicyTemplate is enforced and Template.NamespacedPolicy is true, we would need to create a Policy resource in each target cluster in a particular namespace. In our current implementation the default-policy-controller is not creating a Binding, which is a problem because it's "failing silently". 
How much sense does it make to allow for this? We can also use KubermaticConfiguration for specifying a default global namespace, but would it make sense to have a PolicyTemplate which is enforced and has a namespace (I want to disable this with the existing template webhook). Enforced templates should be for ClusterPolicies IMO. WDYT?

And for the cleanup question, it will be managed by policy-installation-controller, that controller will reconcile PolicyBindings in the UC namespace and take care of cleaning up resources.